src/networkcloud/azext_networkcloud/operations/virtualmachine/_create.py

```diff
@@ -0,0 +1,147 @@
+
+        args_schema.ssh_key_values = AAZListArg(
+            options=["--ssh-key-values"],
+            arg_group="Authentication",
+            help="The list of space-separated SSH public keys."
+        )
+        args_schema.ssh_key_values.Element = AAZStrArg()
+
+        # deregister the VM cli arguments which users should not interact with
+        args_schema.ssh_public_keys._registered = False
+        args_schema.cloud_services_network_attachment.ip_allocation_method._required = False
+        args_schema.cloud_services_network_attachment.ip_allocation_method._registered = False
+
+        return args_schema
+
+    @register_callback
+    def pre_operations(self):
+        args = self.ctx.args
+
+        # Dynamic is the default allocation method for the cloud services network attachment
+        args.cloud_services_network_attachment.ip_allocation_method = "Dynamic"
+
+        ssh_keys = []
+        if bool(args.generate_ssh_keys):
+            ssh_keys += generate_ssh_keys()
+        if list(args.ssh_dest_key_path):
+            ssh_keys += get_ssh_keys_from_path(list(args.ssh_dest_key_path))
+        if list(args.ssh_key_values):
+            ssh_keys += add_ssh_key_action(list(args.ssh_key_values))
+        if len(ssh_keys) == 0:
+            print('No keys are selected for insertion into the vm. The image will need to have keys or credentials setup in order to access.')
+        args.ssh_public_keys = ssh_keys
+
+        return args
+
+
+def generate_ssh_keys():
+    ''' generates ssh keys in user .ssh folder '''
+
+    key_name = "id_rsa"
+    private_key_path = os.path.join(os.path.expanduser('~'), '.ssh', key_name)
+    public_key_path = os.path.join(
+        os.path.expanduser('~'), '.ssh', f"{key_name}.pub")
+    print(f"Attempted to find or generate SSH key files {key_name} and {key_name}.pub under ~/.ssh to allow SSH access "
+          "to the vm. If using machines without permanent storage, back up your keys to a safe location.")
+    return [{'keyData': keys.generate_ssh_keys(private_key_path, public_key_path)}]
+
+
+def get_ssh_keys_from_path(values):
+    ''' get ssh keys from the provided path '''
+
+    key_list = []
+    for path in values:
+        # Casting is necessary to get away from the custom type AAZSimpleType wrapping the string we want
+        path = str(path)
+        if os.path.isdir(path):
+            content = ""
+            for key in os.listdir(path):
+                if key.endswith(".pub"):
+                    with open(os.path.join(path, key,), 'r', encoding="utf-8") as k:
+                        content = k.read()
+                    try:
+                        if keys.is_valid_ssh_rsa_public_key(content):
+                            key_list.append({'keyData': content})
+                    except Exception as exception:
+                        raise CLIError(
+                            f'Unsupported Key {key} is provided.\nContent:\n{content}.') from exception
```

Comment:
`CLIError` is not recommended to raise. Please use error types in https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md

Reply:
Updated